### PR TITLE
metrics for s3

### PR DESCRIPTION
**Pull request by sjamgade**
commit 2 and 3 are essentially the same. commit 3's approach feels less intrusive. 
Commit messages have more info

### CODE REVIEW

**Review by rominf — COMMENTED** [review]
I like a decorator approach more. LGTM, however, I have no enough context to approve it.

File: rohmu/object_storage/s3.py [hunk 1]
```diff
@@ -412,6 +424,7 @@ def multipart_upload_file_object(
             start_of_part_upload = time.monotonic()
             while True:
                 attempts -= 1
+                self.stats.operation(StorageOperation.store_file_from_memory)
```
[the remark this answers is not in the document]
Comment (sjamgade) [reply rohmu/object_storage/s3.py]: added

File: rohmu/object_storage/azure.py [hunk 2]
```diff
@@ -76,9 +77,10 @@ def __init__(
         azure_cloud=None,
         proxy_info=None,
         notifier: Optional[Notifier] = None,
+        statsd_info: Optional[StatsdConfig] = None,
```
Comment (akudiyar) [inline rohmu/object_storage/azure.py:80]: q: This field name looks a bit controversial to me. Why is it "info" and not "config"?

File: rohmu/object_storage/base.py [hunk 3]
```diff
@@ -41,14 +42,15 @@ class Config:
...
-    def __init__(self, prefix, notifier: Optional[Notifier] = None) -> None:
+    def __init__(self, prefix, notifier: Optional[Notifier] = None, statsd_info: Optional[StatsdConfig] = None) -> None:
         self.log = logging.getLogger(self.__class__.__name__)
         if not prefix:
             prefix = ""
         elif prefix[-1] != "/":
             prefix += "/"
         self.prefix = prefix
         self.notifier = notifier or NullNotifier()
+        self.stats = StatsClient(statsd_info)
```
Comment (akudiyar) [inline rohmu/object_storage/base.py:53]: if `None` is passed to `parse_obj()`, an error like `expected dict not NoneType (type=type_error)` will be thrown

**Review by akudiyar — CHANGES_REQUESTED** [review]
Two small comments

File: rohmu/object_storage/azure.py (hunk 2 above)
Comment (sjamgade) [reply rohmu/object_storage/azure.py:80]: we have proxy_info, so went with same things

File: rohmu/object_storage/base.py (hunk 3 above)
Comment (sjamgade) [reply rohmu/object_storage/base.py:53]: StatsClient checks for statsd_info being `None` before proceding

**Review by akudiyar — APPROVED** [review]

**Pull request merged**